packages/store/src/selectors/selector-utils.ts

```diff
@@ -84,29 +40,24 @@ export function createSelector<T extends (...args: any[]) => any>(
       }
     };
   };
-
-  selectorMetaData.makeRootSelector = makeRootSelector;
-
-  return memoizedFn;
 }
 
-function setupSelectorMetadata<T extends (...args: any[]) => any>(
+export function createMemoizedSelectorFn<T extends (...args: any[]) => any>(
   originalFn: T,
-  creationMetadata: CreationMetadata | undefined
+  creationMetadata: Partial<CreationMetadata> | undefined
 ) {
-  const selectorMetaData = ensureSelectorMetadata(originalFn);
-  selectorMetaData.originalFn = originalFn;
-  let getExplicitSelectorOptions = () => ({});
-  if (creationMetadata) {
-    selectorMetaData.containerClass = creationMetadata.containerClass;
-    selectorMetaData.selectorName = creationMetadata.selectorName;
-    getExplicitSelectorOptions =
-      creationMetadata.getSelectorOptions || getExplicitSelectorOptions;
-  }
-  const selectorMetaDataClone = { ...selectorMetaData };
-  selectorMetaData.getSelectorOptions = () =>
-    getLocalSelectorOptions(selectorMetaDataClone, getExplicitSelectorOptions());
-  return selectorMetaData;
+  const containerClass = creationMetadata && creationMetadata.containerClass;
```

Comment:
We can use new features:
```suggestion
  const containerClass = creationMetadata?.containerClass;
```